lib/location/location_utils.c

```diff
@@ -115,3 +118,28 @@ const char *location_utils_nrf_cloud_jwt_generate(void)
 
 	return jwt_buf;
 }
+
+void location_utils_systime_to_location_datetime(struct location_datetime *datetime)
+{
+	struct timespec tp;
+	struct tm ltm = { 0 };
+
+	__ASSERT_NO_MSG(datetime != NULL);
+
+	clock_gettime(CLOCK_REALTIME, &tp);
```

Comment:
Although modem_shell and the Location sample both use the date_time library, it's not mandatory to do so. If date_time library isn't used, the time returned by `clock_gettime()` will be invalid. Maybe this code should be flagged with `CONFIG_DATE_TIME`?

Reply:
That was how I did it 1st, but then started to think the possibility that the system time is set otherwise, without datetime lib. That's why I ended to this solution.

Reply:
Ok, that is true. It's just that if the application doesn't use date_time library or set the system time, the timestamp will be invalid. And this doesn't seem to be documented anywhere.

Reply:
True, we don't know if the system time has been set or is it in 70's. Maybe better to revert to flagging with CONFIG_DATE_TIME. I'll change this.

Reply:
Changed so that valid flag will be enabled only if date_time lib is in use.